tests/helpers_test.go

```diff
@@ -60,60 +72,98 @@ func setupAPI(ctx context.Context, db *storage.Storage) (*service.APIService, er
 	return api, nil
 }
 
-// NewTestClient creates a new API client for testing.
-func NewTestClient(port int) (*client.HTTPclient, error) {
-	return client.New(fmt.Sprintf("http://127.0.0.1:%d", port))
-}
+// setupWeb3 sets up the web3 contracts for testing. It deploys the contracts
+// if the environment variables are not set, if they are set it loads the
+// contracts from the environment variables. It returns the contracts object.
+func setupWeb3(t *testing.T, ctx context.Context) *web3.Contracts {
+	// Get the environment variables
+	var (
+		privKey             = os.Getenv(privKeyEnvVarName)
+		rpcUrl              = os.Getenv(rpcUrlEnvVarName)
+		orgRegistryAddr     = os.Getenv(orgRegistryEnvVarName)
+		processRegistryAddr = os.Getenv(processRegistryEnvVarName)
+		resultsRegistryAddr = os.Getenv(resultsRegistryEnvVarName)
+		zkVerifierAddr      = os.Getenv(zkVerifierEnvVarName)
+	)
+	// Check if the environment variables are set to run the tests over local
+	// geth node or remote blockchain environment
+	localEnv := privKey == "" || rpcUrl == "" || orgRegistryAddr == "" ||
+		processRegistryAddr == "" || resultsRegistryAddr == "" || zkVerifierAddr == ""
+	if localEnv {
+		// Generate a random port for geth HTTP RPC
+		gethPort := util.RandomInt(10000, 20000)
+		rpcUrl = fmt.Sprintf("http://localhost:%d", gethPort)
+		// Set environment variables for docker-compose in the process environment
+		composeEnv := make(map[string]string)
+		composeEnv["GETH_PORT_8545"] = fmt.Sprintf("%d", gethPort)
```

Comment:
shouldnt this constants be called GETH_PORT_RPC_HTTP, GETH_PORT_RPC_WS & GETH_PORT_P2P respectively ?
seems more clear specially if you are using random ports that will not match the ENV names